### PR TITLE
CSCQVAIN-63: Adds Publish button to Datasets view.

### PR DESCRIPTION
**Pull request by jppiiroinen**
This adds the publish button to Datasets view.

### CODE REVIEW

File: src/views/Datasets.vue [hunk 1]
```diff
@@ -49,18 +49,21 @@
...
-							<font-awesome-icon icon="history" fixed-width />Versions
-						</b-dropdown-item-button>
-
-						<b-dropdown-item-button :disabled="row.item.published" size="sm" variant="danger" @click="itemToBeDeleted = row.item.id" v-b-modal.deleteModal>
-							<font-awesome-icon icon="trash" fixed-width />Delete
-						</b-dropdown-item-button>
-					</b-dropdown>
+					<b-button-group>
+						<b-button variant="primary" size="sm" @click.stop="open(row.item.id)"><font-awesome-icon icon="pen" fixed-width />Edit</b-button>
+						<b-button v-if="!row.item.published || row.item.synced < row.item.modified" variant="success" size="sm" @click="itemToBePublished = row.item" v-b-tooltip.hover title="Publish makes the latest saved version public." v-b-modal.publishModal>Publish</b-button>
```
Comment (jppiiroinen) [inline src/views/Datasets.vue]: this needs to be updated

File: src/views/Editor.vue [hunk 2]
```diff
@@ -283,14 +249,14 @@ export default {
 				}
 				if (e.response && e.response.data) {
 					this.publishError = e.response.data
-					this.$root.$emit('bv::show::modal', 'publish-modal', this.$refs['dataset-publish-button'])
+					this.$root.$emit('bv::show::modal', 'publishErrorModal', this.$refs['dataset-publish-button'])
 				} else {
 					this.$root.showAlert("Publish failed!", "danger")
 				}
 				// TODO: consider updating Publish modal error boiler plate with this error message
```
Comment (Kauhia) [inline src/views/Editor.vue]: Should these comments be deleted?

File: src/views/Editor.vue [hunk 3]
```diff
@@ -91,29 +91,21 @@
...
 		<dataset-json-modal id="dataset-json-modal"></dataset-json-modal>
 		<dataset-overview-modal id="dataset-overview-modal"></dataset-overview-modal>
-		<publish-modal id="publish-modal" :error="publishError" @hidden="publishError = null"></publish-modal>
+		<b-modal ref="publishModal" id="publishModal" title="Publish dataset?"
+			ok-title="Publish" cancel-variant="primary" ok-variant="success" @ok="publish">
+			<p :style="{'text-align': 'center'}">
+				<p>I understand that publishing this dataset...</p>
+				<ul class="list-unstyled">
+					<li class="font-italic">... will make it available publicly</li>
+					<li class="font-italic">... marks it as ready and enables editing restrictions</li>
```
Comment (Kauhia) [inline src/views/Editor.vue]: Do we want to keep the leading ... before the actual text?

**Review by Kauhia — CHANGES_REQUESTED** [review]

File: src/views/Editor.vue (hunk 2 above)
Comment (jppiiroinen) [reply src/views/Editor.vue]: those should be

Comment (jppiiroinen) [reply src/views/Editor.vue]: fixed

File: src/views/Editor.vue (hunk 3 above)
Comment (jppiiroinen) [reply src/views/Editor.vue]: I will change it

**Review by Kauhia — APPROVED** [review]

**Pull request merged**